test/arrayops.jl

```diff
@@ -1926,3 +1926,32 @@ using TestHelpers.OAs
     @test accumulate(op, [10,20, 30]) == [10, op(10, 20), op(op(10, 20), 30)] == [10, 40, 110]
     @test accumulate(op, [10 20 30], 2) == [10 op(10, 20) op(op(10, 20), 30)] == [10 40 110]
 end
+
+@testset "zeros and ones" begin
+    @test ones([1,2], Float64, (2,3)) == ones(2,3)
+    @test ones(2) == ones(Int, 2) == ones([2,3], Float32, 2) ==  [1,1]
+    @test isa(ones(2), Vector{Float64})
+    @test isa(ones(Int, 2), Vector{Int})
+    @test isa(ones([2,3], Float32, 2), Vector{Float32})
+
+    function test_zeros(arr, T, s)
+        @test all(arr .== 0)
+        @test isa(arr, T)
+        @test size(arr) == s
+    end
+    test_zeros(zeros(2),     Vector{Float64}, (2,))
+    test_zeros(zeros(2,3),   Matrix{Float64}, (2,3))
+    test_zeros(zeros((2,3)), Matrix{Float64}, (2,3))
+
+    test_zeros(zeros(Int, 6),      Vector{Int}, (6,))
+    test_zeros(zeros(Int, 2, 3),   Matrix{Int}, (2,3))
+    test_zeros(zeros(Int, (2, 3)), Matrix{Int}, (2,3))
+
+    test_zeros(zeros([1 2; 3 4]), Matrix{Int64}, (2, 2))
+    test_zeros(zeros([1 2; 3 4], Float64), Matrix{Float64}, (2, 2))
+
+    zs = zeros(SparseMatrixCSC([1 2; 3 4]), Complex{Float64}, (2,3))
+    test_zeros(zs, SparseMatrixCSC{Complex{Float64}}, (2, 3))
+
+    @test_throws Exception zeros(Float64, [1.]) #19265
```

Comment:
should test specifically, otherwise a typo in the implementation could pass this test